src/PIL/ImageOps.py

```diff
@@ -600,22 +601,30 @@ def exif_transpose(image):
     }.get(orientation)
     if method is not None:
         transposed_image = image.transpose(method)
-        transposed_exif = transposed_image.getexif()
-        if 0x0112 in transposed_exif:
-            del transposed_exif[0x0112]
-            if "exif" in transposed_image.info:
-                transposed_image.info["exif"] = transposed_exif.tobytes()
-            elif "Raw profile type exif" in transposed_image.info:
-                transposed_image.info[
-                    "Raw profile type exif"
-                ] = transposed_exif.tobytes().hex()
-            elif "XML:com.adobe.xmp" in transposed_image.info:
+        if inPlace:
+            image.im = transposed_image.im
+            image.pyaccess = None
+            image._size = transposed_image._size
+        exif_image = image if inPlace else transposed_image
+
+        exif = exif_image.getexif()
+        if 0x0112 in exif:
+            del exif[0x0112]
+            if "exif" in exif_image.info:
+                exif_image.info["exif"] = exif.tobytes()
+            elif "Raw profile type exif" in exif_image.info:
+                exif_image.info["Raw profile type exif"] = exif.tobytes().hex()
+            elif "XML:com.adobe.xmp" in exif_image.info:
                 for pattern in (
                     r'tiff:Orientation="([0-9])"',
                     r"<tiff:Orientation>([0-9])</tiff:Orientation>",
                 ):
-                    transposed_image.info["XML:com.adobe.xmp"] = re.sub(
-                        pattern, "", transposed_image.info["XML:com.adobe.xmp"]
+                    exif_image.info["XML:com.adobe.xmp"] = re.sub(
+                        pattern, "", exif_image.info["XML:com.adobe.xmp"]
                     )
+        if inPlace:
+            return
         return transposed_image
+    if inPlace:
```

Comment:
use `else` here along with upmost `if` above to avoid check again when `method is not None`.

such as:
```python
def exif_transpose(image, inPlace=False):
    if method is not None:
        # apply exif orientation
        if not inPlace:
            return transposed_image
    else:
        if not inPlace:  #can merge into `else` line.
            return image.copy()
```